### PR TITLE
Add dark mode

### PR DESCRIPTION
**Pull request by JosephusPaye**
For all the night owls out there :owl:

- [x] Style landing page
- [x] Style content pages
- [x] Add toggle button
- [x] Forgot the 404 page!

[Preview](https://deploy-preview-90--nubook.netlify.app)

### CODE REVIEW

**Comment by KipHamiltons** [comment]
The landing page on the preview is still light mode for me

**Comment by JosephusPaye** [comment]
Right now it only kicks in if you've enabled dark mode in your OS/browser settings. I'm going to add a button to manually toggle it.

If you want to test it, run the following in your browser console on the landing page:

```js
document.documentElement.classList.remove('light')
document.documentElement.classList.add('dark')
```

**Comment by KipHamiltons** [comment]
dang that looks nice!!

File: gatsby-ssr.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/*eslint-env node */
+
+const React = require('react')
+
+// Adds the dark mode class name to the body element before render
```
Comment (JosephusPaye) [inline gatsby-ssr.js:5]:
```suggestion
// Adds the dark mode class name to the body element before render, inline to void a flash of white in dark mode
```

**Comment by JosephusPaye** [comment]
It's done!

File: src/book/kitchen-sink/02-formatting-paragraphs.mdx [hunk 2]
```diff
@@ -25,7 +25,7 @@ _single underscores_
 
 **double asterisks**
 
-This paragraph has some `code` in it.
+This paragraph has some `code` in it. This sentence has a keyboard shortcut: press <kbd>Enter</kbd> to confirm. To deny, press <kbd>Ctrl</kbd>+<kbd>E</kbd>.
```
Comment (KipHamiltons) [inline src/book/kitchen-sink/02-formatting-paragraphs.mdx:28]: This seems unrelated?

**Review by KipHamiltons — COMMENTED** [review]
Looks great!! I love the aesthetic and tones overall. really nice to use. great job! I tried a few pages with a variety of elements on chrome/firefox/safari on macos, and they all looked good without any broken elements that I found.

I can't really review the js, of course.

One very small irk I have is that the code snippet background does is very low contrast with the page background. This isn't a big deal, because the beautiful red tone of the monospace font makes the code snippets quite obvious, but I think they would benefit (slightly) with slightly more contrast.

**Comment by ysims** [comment]
Light mode styling change looks nice, and more inline with the landing page. Although, should it really be in this PR? 😛 
Is there a toggle somewhere I'm missing for dark mode or is it just from the browser?

File: src/book/kitchen-sink/02-formatting-paragraphs.mdx (hunk 2 above)
Comment (JosephusPaye) [reply src/book/kitchen-sink/02-formatting-paragraphs.mdx:28]: There's no place in the kitchen sink where we test `<kbd>`, and it actually belongs here on the inline formatting page.

**Comment by JosephusPaye** [comment]
@ysims:

> Although, should it really be in this PR? 😛

That's true. I did the version for dark mode then decided to update light mode too, since it would be weird having the two different styles until it's fixed in another PR.

> Is there a toggle somewhere I'm missing for dark mode or is it just from the browser?

Toggle's at the bottom right in the footer:

image

Couldn't put it in the header as there isn't enough space to keep it there on mobile.

**Comment by JosephusPaye** [comment]
@KipHamiltons I'll increase the inline code background contrast a bit in dark mode.

**Review by tayloryoung6396 — APPROVED** [review]
LGTM

**Review by KipHamiltons — APPROVED** [review]
